ui/packages/catalog-ui-search/src/main/webapp/js/CQLUtils.js

```diff
@@ -9,254 +9,295 @@
-        },
-        polygonToCQLMultiPolygon: function(model) {
-            return model.map(this.polygonToCQLPolygon);
-        },
-        lineToCQLLIne: function(model){
-            var cqlLINE = model.map(function (point) {
-                return point[0] + ' ' + point[1];
-            });
-            return cqlLINE;
-        },
-        isGeoFilter: function(type){
-            return (type === 'DWITHIN' || type === 'INTERSECTS');
-        },
-        transformFilterToCQL: function(filter){
-            return this.sanitizeGeometryCql("(" + cql.write(cql.simplify(cql.read(cql.write(filter)))) + ")");
-        },
-        transformCQLToFilter: function(cqlString){
-            return cql.simplify(cql.read(cqlString));
-        },
-        isPointRadiusFilter: function(filter){
-            return filter.value && filter.value.value && filter.value.value.indexOf('POINT') >= 0;
-        },
-        buildIntersectCQL: function(locationGeometry){
-            var locationFilter = "";
-            var locationWkt = locationGeometry.toWkt();
-            var locationType = locationGeometry.toGeoJSON().type.toUpperCase();
+    }
+}
+
+function generateFilterForFilterFunction(filterFunctionName, params) {
+    return {
+        type: '=',
+        value: true,
+        property: {
+            type: 'FILTER_FUNCTION',
+            filterFunctionName,
+            params
+        }
+    };
+}
 
-            var shapes;
-            switch (locationType) {
-              case "POINT":
-              case "LINESTRING":
-                    locationFilter = "(DWITHIN(anyGeo, " + locationWkt + ", 1, meters))"
-                    break;
-              case "POLYGON":
-                    // Test if the shape wkt contains ,(
-                    if (/,\(/.test(locationWkt)) {
-                        shapes = locationWkt.split(',(');
+function isGeoFilter(type) {
+    return (type === 'DWITHIN' || type === 'INTERSECTS');
+}
 
-                        $.each(shapes, function (i, polygon) {
-                            locationWkt = polygon.replace(/POLYGON|[()]/g, '');
-                            locationWkt = "POLYGON((" + locationWkt + "))"
-                            locationFilter += "(INTERSECTS(anyGeo, " + locationWkt + "))";
+function transformFilterToCQL(filter) {
+    return this.sanitizeGeometryCql("(" + cql.write(cql.simplify(cql.read(cql.write(filter)))) + ")");
+}
 
-                            if (i !== shapes.length - 1) {
-                                  locationFilter += " OR ";
-                            }
-                        }.bind(this));
-                    }
-                    else {
-                        locationFilter = "(INTERSECTS(anyGeo, " + locationWkt + "))";
+function transformCQLToFilter(cqlString) {
+    return cql.simplify(cql.read(cqlString));
+}
+
+function isPointRadiusFilter(filter) {
+    return filter.value && filter.value.value && filter.value.value.indexOf('POINT') >= 0;
```

Comment:
❓ is there any issue if we have a Multi Geometry WKT containing a POINT geo?

Reply:
That's a good question, but this logic is the same as before so I don't think there are any issues with it. I'm leaning towards if it ain't broke, don't fix it.